### PR TITLE
167_two_sum

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code

### CODE REVIEW

File: 167_two_sum.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+
+class Solution:
+    def twoSum(self, numbers: List[int], target: int) -> List[int]:
```
Comment (ryujumoto) [inline 167_two_sum.py:8]: This solution is wrong, so correct it to be accepted.

**Review by ryujumoto — COMMENTED** [review]
Check my review.

File: 167_two_sum.py (hunk 1 above)
Comment (Rwatana) [reply 167_two_sum.py:5]: Let me ask an question. Which mistake did I make a grammatical mistake or a method's problem.

Comment (ryujumoto) [reply 167_two_sum.py:8]: You made the latter mistake. I think this code won't be accepted on LeetCode.

**Comment by Rwatana** [comment]
Let me ask an question.
The below is the code which I made by using binary trees. in my opinion, It is better because the old one is T(logn)O(1) but, the new one is T(nlogn)O(1) . Is it True?

```
def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            new_target= target-numbers[i]
            left , right = i+1, len(numbers)-1
            while left <= right:
                middle = (left+right)//2
                if numbers[middle] > new_target:
                    right = middle-1
                elif numbers[middle] < new_target:
                    left = middle+1
                else:
                    return [i+1,middle+1]
```
@Ryuju0911

**Comment by ryujumoto** [comment]
> Let me ask an question. The below is the code which I made by using binary trees. in my opinion, It is better because the old one is T(logn)O(1) but, the new one is T(nlogn)O(1) . Is it True?
> 
> ```
> def twoSum(self, numbers: List[int], target: int) -> List[int]:
>         for i in range(len(numbers)):
>             new_target= target-numbers[i]
>             left , right = i+1, len(numbers)-1
>             while left <= right:
>                 middle = (left+right)//2
>                 if numbers[middle] > new_target:
>                     right = middle-1
>                 elif numbers[middle] < new_target:
>                     left = middle+1
>                 else:
>                     return [i+1,middle+1]
> ```
> 
> @Ryuju0911

The time complexity and space complexity are as following.

- The old one: time: O(n), space: O(1)

- The new one: time: O(nlogn), space: O(1)

The old one is better, but this is a practice of binary search, so I wanted you to write a binary search code, too.

File: 167_two_sum.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            new_target= target-numbers[i]
+            left , right = i+1, len(numbers)-1
+            while left <= right:
+                middle = (left+right)//2
+                if numbers[middle] > new_target:
+                    right = middle-1
+                elif numbers[middle] < new_target:
+                    left = middle+1
+                else:
+                    return [i+1,middle+1]
```
Comment (ryujumoto) [inline 167_two_sum.py]:
```
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            new_target = target-numbers[i]
            left , right = i + 1, len(numbers) - 1
            while left <= right:
                middle = (left+right) // 2
                if numbers[middle] > new_target:
                    right = middle - 1
                elif numbers[middle] < new_target:
                    left = middle + 1
                else:
                    return [i + 1,middle + 1]
```

**Comment by Rwatana** [comment]
I changed. Please check my code.

**Review by ryujumoto — APPROVED** [review]

**Pull request merged**